test/cpp/naming/resolver_component_test.cc

```diff
@@ -192,9 +192,9 @@ gpr_timespec TestDeadline(void) {
 
 struct ArgsStruct {
   gpr_event ev;
-  gpr_atm done_atm;
   gpr_mu* mu;
```

Comment:
Can we convert this to use `grpc_core::Mutex` instead, so that we can use lock annotations and allow the compiler to enforce the lock guard?

We can do the same thing in the tests that were fixed in #25398.

Reply:
I'd like to do that, but we can't because this `gpr_mu` actually belongs to the pollset, so we'd need to migrate pollset to `grpc_core::Mutex` first.

Reply:
Ah, okay.  This is probably something that we'll be able to address as part of migrating this code to EventEngine.